perun-core/src/main/java/cz/metacentrum/perun/core/impl/GroupsManagerImpl.java

```diff
@@ -575,20 +600,28 @@ public int getSubGroupsCount(PerunSession sess, Group parentGroup) {
 		namedParams.addValue("offset", query.getOffset());
 		namedParams.addValue("limit", query.getPageSize());
 
+		if (query.getMemberId() != null) namedParams.addValue("memberId", query.getMemberId());
+
+		String selectQuery = getSQLSelectForGroupsPage(query);
 		String searchQuery = getSQLWhereForGroupsPage(query, namedParams);
 
-		return namedParameterJdbcTemplate.query(
-			"SELECT " + groupMappingSelectQuery +
-			", count(*) OVER() AS total_count" +
-			" FROM groups" +
-			" WHERE groups.vo_id=(:voId)" +
-			" AND groups.id IN (:groupsIds)" +
-			searchQuery +
-			" ORDER BY " + query.getSortColumn().getSqlOrderBy(query) +
-			" OFFSET (:offset)" +
-			" LIMIT (:limit);",
+		Paginated<Group> groups = namedParameterJdbcTemplate.query(
+			selectQuery +
+				" WHERE groups.vo_id=(:voId)" +
+				" AND groups.id IN (:groupsIds)" +
+				searchQuery +
+				" ORDER BY " + query.getSortColumn().getSqlOrderBy(query) +
+				" OFFSET (:offset)" +
+				" LIMIT (:limit);",
 			namedParams,
 			getPaginatedGroupsExtractor(query));
+
+		// remove "members" group
+		if(groups != null && !groups.getData().isEmpty() && query.getMemberId() != null) {
+			groups.getData().removeIf(g -> g.getName().equals(VosManager.MEMBERS_GROUP));
```

Comment:
Filtering members group here is too late, because the page size is given by the LIMIT parameter in the sql query and removing one of the groups after performing the query would return result not fitting the page size. You should put this  condition into the query.